### PR TITLE
migrate vendor to golang/dep

### PR DESCRIPTION
**Pull request by nogoegst**
Imprortant notice: this PR updates vendor of following packages:
 * github.com/docker/docker
 * golang.org/x/sys
 * google.golang.org/api

Probably removing of obsolete godep files should be here as well.

### CODE REVIEW

**Comment by dgageot** [comment]
😍

**Comment by dgageot** [comment]
@nogoegst Please ping me when it's ready to review

**Comment by nogoegst** [comment]
@dgageot, I think it is ready for review. :wink:

**Comment by nogoegst** [comment]
I think that all corresponding changes to the docs should be here as well. I'll add them a bit later.

**Comment by dgageot** [comment]
+ `make build` works
+ `USE_CONTAINER=true make build` works

**Comment by dgageot** [comment]
@nogoegst Can you update those instructions too? https://github.com/docker/machine/blob/master/CONTRIBUTING.md#restore-update-and-save-dependencies

**Comment by dgageot** [comment]
@nogoegst Forget about it. It's just me being drunk

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -145,23 +145,11 @@ Scrub build results:
...
-   This uses godep to restores all the dependencies to your `$GOPATH`. Note that this changes the packages in your `$GOPATH`
-
-#### Add one ore more dependencies
-
-    go get -u <new dependency>
-
-#### Save the dependencies to `vendor/`
-
-    make dep-save
+When you make a fresh copy of the repo, all the dependecies are in `vendor/` directory for the builds to work.
```
Comment (dgageot) [inline CONTRIBUTING.md]: s/dependecies/dependencies/

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -145,23 +145,11 @@ Scrub build results:
...
+This project uses [golang/dep](https://github.com/golang/dep) as vendor management tool. Please refer to `dep` documentation
+on further details.
```
Comment (dgageot) [inline CONTRIBUTING.md]: s/on/for/

File: CONTRIBUTING.md (hunk 1 above)
Comment (dgageot) [inline CONTRIBUTING.md]: s/builds/build/

**Comment by dgageot** [comment]
@nogoegst Please fix the typos and then I think we are good to go \o/

File: CONTRIBUTING.md (hunk 1 above)
Comment (nogoegst) [reply CONTRIBUTING.md]: btw there are multiple options to build I guess. That's why there was 'builds'.

**Comment by nogoegst** [comment]
@dgageot fixed, thanks!

**Comment by dgageot** [comment]
LGTM

**Comment by dgageot** [comment]
@vdemeester or @shin- would you have a moment to review?

**Comment by dgageot** [comment]
@vdemeester Thanks!

**Pull request merged**